python_ta/cfg/cfg_generator.py

```diff
@@ -13,6 +13,8 @@
 from astroid import nodes
 from astroid.builder import AstroidBuilder
 
+from python_ta.transforms.z3_visitor import Z3Visitor
```

Comment:
As with my previous comments, here you should use a relative import (`.transforms.z3_visitor`) and group this with the imports below.

But also similar to `graph.py`, we need to check for whether Z3 has been successfully imported or not, and if not then make sure the previous functionality works correctly. You can test this by uninstalling Z3 from your Python environment and then calling `generate_cfg`.

Reply:
Hi Professor, I tested that when Z3 is uninstalled, `z3_visitor` would raise an error as we did not check for Z3 dependency in its imports. I suggest we should modify `z3_visitor` such that if Z3 dependency is not available, invoking the visitor would do nothing. This may be more convenient than checking for Z3 dependency whenever we want to use `Z3Visitor`. Should I make this change in a separate PR or include it in this PR.

Reply:
Hi @Raine-Yang-UofT, please use the approach I described in my original comment. I don't agree with the approach of `Z3Visitor` "doing nothing" if `z3` isn't installed, I'd much rather it not be invoked at all.

Reply:
@david-yz-liu No problem, I have added a check for z3 dependency in generating CFG. Now the CFG generation works correctly even with Z3 uninstalled. However, as the import of z3 are not checked in `Z3Visitor` and `Z3Parser`, we need to fix these two modules for PythonTA to function correctly without Z3. Let me know if I should propose this change in a new PR.